src/clj/game/core/runs.clj

```diff
@@ -4,30 +4,224 @@
          gain-run-credits update-ice-in-server update-all-ice
          get-agenda-points gain-agenda-point optional-ability
          get-remote-names card-name can-access-loud can-steal?
-         prevent-jack-out card-flag? can-run?)
+         prevent-jack-out card-flag? can-run? run-approach-ice
+         jack-out successful-run end-run update-step corp-phase-53)
 
 ;;; Steps in the run sequence
+(defn approach-server
+  "[5]: the runner approaches the attacked server"
+  [state side eid]
+  ;; "When approached" conditionals meet their trigger condition
+  (prn "approach-server")
+  (trigger-event-sync state side (make-eid state) :approach-server (-> @state :run :server))
+  ;; [5.2] the runner chooses whether to jack out or declare a successful run
+  ; (show-wait-prompt state :corp "Runner actions. Paid abilities")
+  (wait-for (resolve-ability
+              state :runner
+              {:prompt "Complete a successful run, or jack out? Paid abilities"
+               :choices ["Successful run" "Jack out"]
+               :prompt-type :run
+               :priority -1
+               :async true
+               :effect (req ;(clear-wait-prompt state :corp)
+                            (if (= "Jack out" target)
+                              (update-step state side eid :step5 :step7)
+                              (effect-completed state side eid)))}
+              nil nil)
+            (when-not (:ended (:run @state))
+              ; (show-wait-prompt state :runner "Corp may rez non-ice. Paid abilities")
+              (wait-for (resolve-ability
+                          state :corp
+                          {:prompt "Act before access?"
+                           :choices ["Action before access" "No more actions"]
```

Comment:
I like the current way of handling the Corp's "act after commit to access" window, those this mean that you will end up with an extra prompt after runner selects "successful run" in the case the corp does not want to act?

Reply:
Yeah, it's not super graceful right now. I wanted to fully remove the `corp-phase-53` system, but _a lot_ of stuff rests on it, so this was my stop-gap.